tests/s3/copy_object/test_copy_obj_simultaneous_del.py

```diff
@@ -137,6 +155,45 @@ def create_put_parallel_copy_and_delete_object(self, copy_args, delete_args, is_
                 process2 = pool.apply_async(self.delete_object_wrapper, args=delete_args)
                 assert_utils.assert_true(process1.get()[0], process1.get()[1])
                 assert_utils.assert_true(process2.get()[0], process2.get()[1])
+            if copy_args[0] == copy_args[2] and copy_args[1] == copy_args[3]:
+                self.head_object_wrapper(copy_args[0], copy_args[1], "404")
+
+    def create_put_parallel_copy_and_delete_bucket(self, copy_args, delete_args, is_mpu=False):
+        """Create and upload object to bucket and then Parallel copy object and delete bucket with
+        given arguments"""
+        for obj_size in [1024, 5*1024]:
+            if obj_size == 1024:
+                LOGGER.info("Upload a large object (~1GB) srcobj to srcbuck")
+            else:
+                LOGGER.info("Upload a large object (~5GB) srcobj to srcbuck")
+            if is_mpu:
+                resp = system_utils.create_file(fpath=copy_args[6], count=obj_size, b_size="1M")
+                assert_utils.assert_true(resp[0], resp[1])
+                resp = self.s3mp_obj.simple_multipart_upload(copy_args[0], copy_args[1], obj_size,
+                                                             copy_args[6], copy_args[5])
+            else:
+                self.create_put_object(copy_args[0], copy_args[1], obj_size)
+            LOGGER.info("Parallely, \n1. From %s copy %s to %s as %s\n2. Delete %s",
+                        copy_args[0], copy_args[1], copy_args[2], copy_args[3], delete_args[0])
+            with multiprocessing.Pool(processes=2) as pool:
+                process1 = pool.apply_async(self.copy_object_wrapper, args=copy_args[:5])
+                process2 = pool.apply_async(self.delete_bucket_wrapper, args=delete_args)
+                assert_utils.assert_true(process1.get()[0], process1.get()[1])
+                assert_utils.assert_true(process2.get()[0], process2.get()[1])
```

Comment:
@apoorva-rao Can't we create some generic function to support this requirements? cc: @VikasKumarC

Reply:
Will be doing code optimization in other PR as part of ticket - CORTX-33468